src/tea_json.res

```diff
@@ -222,35 +222,53 @@ module Decoder = {
 
   let map = (mapper, Decoder(decoder1)) => Decoder(
     value => {
-      open Tea_result
       switch decoder1(value) {
       | Ok(v1) => Ok(mapper(v1))
       | Error(e) => Error("map " ++ e)
       }
     },
   )
 
+let error = x =>
```

Comment:
I think these would maybe be clearer if they was inlined into where they are used?